daml/ContingentClaims/Math/Stochastic.daml

```diff
@@ -136,16 +138,19 @@ fapf ccy disc exch today = flip evalState 0 . futuM coalg . Left . (, today) whe
   ϵ (One asset, t) = pure $ exch' asset ccy t
   ϵ (Give c, t) = pure . NegF . claim $ (c, t)
   ϵ (Scale k c, t) = pure $ obs (k, t) `MulF` claim (c, t)
-  ϵ (And c c', t) = pure $ claim (c, t) `AddF` claim (c', t)
-  ϵ (Or c c', t) = pure $ AddF
-      (ind (claim (c , t)) (claim (c', t)) * claim (c , t))
+  ϵ (And cs, t) = pure . SumF $ fmap (claim . (, t)) cs 
+  ϵ (Or [c, c'], t) = pure $ SumF [
```

Comment:
Need to handle multiple-element lists.

This is achieved by permuting the indicator function as if it were a 'binary' number.
Possibly consider using a combinatorial representation here e.g. `nCr` here instead of expanding into a huge sum of indicator functions, which will quickly grow large.